### PR TITLE
Add test for parsing connection paths

### PR DESCRIPTION
**Pull request by Reecepbcups**
## Description

Add test for parsing of connection path.

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by charleenfei — APPROVED** [review]
one small req otherwise lgtm! thanks!

File: modules/core/24-host/parse.go [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 import (
+	"regexp"
 	"strconv"
 	"strings"
 
 	errorsmod "cosmossdk.io/errors"
 )
 
+var pathRegex = regexp.MustCompile("[^/]+")
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply modules/core/24-host/parse.go:18]: Resolved in 5f8d39a37e6fcfd6155eef9e0b6e192a733c783e

**Comment by crodriguezvega** [comment]
@Reecepbcups We discussed internally this PR and [we'd rather keep using `strings.Split`](https://github.com/cosmos/ibc-go/issues/63#issuecomment-1691974727). However, we would like to still merge a part of this PR: [the `TestMustParseConnectionPath` you have added](https://github.com/cosmos/ibc-go/pull/4111/files#diff-1afa83a983010bf65df267e96606242213ff89079c52a739f2654a11513c6fa6). Would it be possible for you to revert the other changes, but leave the test (it should still pass)? I am happy to do this as well, if you're busy. Thank you!

**Comment by Reecepbcups** [comment]
@crodriguezvega makes sense, completed

**Review by damiannolan — APPROVED** [review]
Nice one @Reecepbcups! Thank you

**Review by crodriguezvega — APPROVED** [review]
Thank you, @Reecepbcups!

**Pull request merged**